rules/models/changes/ruleset_test.go

```diff
@@ -1,235 +1,169 @@
-	testBytes = b
-
-	var op UpdateOp
-	err = op.UnmarshalJSON(testBytes)
-	require.Error(t, err)
+	ruleSet.Sort()
+	require.Equal(t, expected, ruleSet)
+	require.Equal(t, expected, ruleSet)
 }
 
 var (
-	ruleSets = RuleSetChangeSets{
-		RuleSetChangeSet{
-			Namespace: "service3",
-		},
-		RuleSetChangeSet{
-			Namespace: "service2",
+	ruleSet = RuleSetChanges{
+		Namespace: "service1",
+		MappingRuleChanges: []MappingRuleChange{
+			{
+				Op:     ChangeOp,
+				RuleID: p("mrID1"),
+				RuleData: &models.MappingRule{
+					Name: "change3",
+				},
+			},
+			{
+				Op: AddOp,
+				RuleData: &models.MappingRule{
+					Name: "Add2",
+				},
+			},
+			{
+				Op:     ChangeOp,
+				RuleID: p("mrID2"),
+				RuleData: &models.MappingRule{
+					Name: "change1",
+				},
+			},
+			{
+				Op:     ChangeOp,
+				RuleID: p("mrID3"),
+				RuleData: &models.MappingRule{
+					Name: "change2",
+				},
+			},
+			{
+				Op: AddOp,
+				RuleData: &models.MappingRule{
+					Name: "Add1",
+				},
+			},
 		},
-		RuleSetChangeSet{
-			Namespace: "service1",
-			MappingRuleChanges: []MappingRuleChange{
-				{
-					Op:     ChangeOp,
-					RuleID: "mrID1",
-					RuleData: &models.MappingRule{
-						Name: "change3",
-					},
-				},
-				{
-					Op: AddOp,
-					RuleData: &models.MappingRule{
-						Name: "Add2",
-					},
-				},
-				{
-					Op:     ChangeOp,
-					RuleID: "mrID2",
-					RuleData: &models.MappingRule{
-						Name: "change1",
-					},
-				},
-				{
-					Op:     ChangeOp,
-					RuleID: "mrID3",
-					RuleData: &models.MappingRule{
-						Name: "change2",
-					},
-				},
-				{
-					Op: AddOp,
-					RuleData: &models.MappingRule{
-						Name: "Add1",
-					},
-				},
-			},
-			RollupRuleChanges: []RollupRuleChange{
-				{
-					Op:     ChangeOp,
-					RuleID: "rrID1",
-					RuleData: &models.RollupRule{
-						Name: "change3",
-					},
-				},
-				{
-					Op: AddOp,
-					RuleData: &models.RollupRule{
-						Name: "Add2",
-					},
-				},
-				{
-					Op:     ChangeOp,
-					RuleID: "rrID2",
-					RuleData: &models.RollupRule{
-						Name: "change1",
-					},
-				},
-				{
-					Op:     ChangeOp,
-					RuleID: "rrID3",
-					RuleData: &models.RollupRule{
-						Name: "change2",
-					},
-				},
-				{
-					Op: AddOp,
-					RuleData: &models.RollupRule{
-						Name: "Add1",
-					},
+		RollupRuleChanges: []RollupRuleChange{
+			{
+				Op:     ChangeOp,
+				RuleID: p("rrID1"),
+				RuleData: &models.RollupRule{
+					Name: "change3",
+				},
+			},
+			{
+				Op: AddOp,
+				RuleData: &models.RollupRule{
+					Name: "Add2",
+				},
+			},
+			{
+				Op:     ChangeOp,
+				RuleID: p("rrID2"),
+				RuleData: &models.RollupRule{
+					Name: "change1",
+				},
+			},
+			{
+				Op:     ChangeOp,
+				RuleID: p("rrID3"),
+				RuleData: &models.RollupRule{
+					Name: "change2",
+				},
+			},
+			{
+				Op: AddOp,
+				RuleData: &models.RollupRule{
+					Name: "Add1",
 				},
 			},
 		},
 	}
 )
+
+func p(s string) *string {
```

Comment:
nit: maybe call this `ptr`, a bit worried that if someone actually did `fmt.Printf(p)` and didn't define p, this function would get printed.